coderd/workspacebuilds_test.go

```diff
@@ -401,11 +401,12 @@ func TestPatchCancelWorkspaceBuild(t *testing.T) {
 		require.Eventually(t, func() bool {
 			var err error
 			build, err = client.WorkspaceBuild(ctx, build.ID)
+			// job gets marked Failed when there is an Error; in practice we never get to Status = Canceled
+			// because provisioners report an Error when canceled. We check the Error string to ensure we don't mask
+			// other errors in this test.
 			return assert.NoError(t, err) &&
-				// The job will never actually cancel successfully because it will never send a
-				// provision complete response.
-				assert.Empty(t, build.Job.Error) &&
-				build.Job.Status == codersdk.ProvisionerJobCanceling
+				build.Job.Error == "canceled" &&
+				build.Job.Status == codersdk.ProvisionerJobFailed
```

Comment:
👍

Reply:
no, assertions will be enforced every time, and will fail the test if they fail even once.  We only care that the error _eventually_ is "canceled", not that it _always_ is.